tests/test_tabular_transform.py

```diff
@@ -0,0 +1,58 @@
+from pandas.core.dtypes.dtypes import CategoricalDtype
+import pytest
+from fastai import *
+from fastai.tabular import *
+
+def test_categorify():
+    path = untar_data(URLs.ADULT_SAMPLE)
+    df = pd.read_csv(path / 'adult.csv')
+    cat_names = ['workclass', 'education', 'marital-status', 'occupation',
+                 'relationship', 'race', 'sex', 'native-country']
+    df = df[cat_names]
+    train_df, valid_df = df[:1024].copy(),df[1024:1260].copy()
+    categorify_transform = Categorify(cat_names, [])
+
+    categorify_transform.apply_train(train_df)
+    categorify_transform.apply_test(valid_df)
+
+    assert all([isinstance(dt, CategoricalDtype) for dt in train_df.dtypes])
+    assert all([isinstance(dt, CategoricalDtype) for dt in valid_df.dtypes])
```

Comment:
Could we keep the whole df and check the columns not in cat_names aren't modified?